src/tools/sssctl/sssctl_logs.c

```diff
@@ -60,19 +71,95 @@ struct sssctl_logs_opts {
     int archived;
 };
 
-errno_t set_debug_level(struct debuglevel_tool_ctx *tool_ctx,
-                        int debug_to_set, const char *config_file)
+struct sssctl_service_desc {
+    const char *name;
+    int mask;
+};
+
+enum serv_idx {
+    SERV_SSSD,
+    SERV_NSS,
+    SERV_PAM,
+    SERV_SUDO,
+    SERV_AUTOFS,
+    SERV_SSH,
+    SERV_PAC,
+    SERV_IFP,
+    SERV_COUNT
+};
+
+struct sssctl_service_desc services[] = {
+    { "sssd",   1U << SERV_SSSD  },
+    { "nss",    1U << SERV_NSS   },
+    { "pam",    1U << SERV_PAM   },
+    { "sudo",   1U << SERV_SUDO  },
+    { "autofs", 1U << SERV_AUTOFS},
+    { "ssh",    1U << SERV_SSH   },
+    { "pac",    1U << SERV_PAC   },
+    { "ifp",    1U << SERV_IFP   }
+};
+
+
+errno_t get_debug_level(TALLOC_CTX *ctx, struct debuglevel_tool_ctx *tool_ctx,
+                        const char *component, int *_value)
+{
+    errno_t ret;
+    uint32_t value;
+    const char *busname;
+    struct sss_domain_info *domain;
+    struct sbus_sync_connection *conn;
+
+    if (strcmp(component, "sssd") == 0) {
+        busname = talloc_strdup(ctx, SSS_BUS_MONITOR);
+    } else if (STARTS_WITH(component, "domain/")) {
+        ret = confdb_get_domain(tool_ctx->confdb,
+                                REMOVE_PREFIX(component, "domain/"), &domain);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "Unknown domain: %s\n", component);
+            goto done;
+        }
+
+        busname = sss_iface_domain_bus(ctx, domain);
+    } else {
+        busname = talloc_asprintf(ctx, "sssd.%s", component);
+    }
+    if (busname == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to create the bus name for %s\n",
+              component);
+        ret = ENOENT;
+        goto done;
+    }
+
+    conn = sbus_sync_connect_private(ctx, SSS_MONITOR_ADDRESS, NULL);
+    if (conn == NULL) {
+        ret = EIO;
+        goto done;
+    }
+
+    ret = sbus_get_service_debug_level(conn, busname, SSS_BUS_PATH, &value);
+    if (ret != EOK) {
+        ret = ENOENT;
+        goto done;
+    }
+
+    *_value = (int) value;
+    ret = EOK;
+
+done:
+
```

Comment:
This function is a candidate for [temporary memory context](https://sssd.io/contrib/talloc.html#temporary-memory-context) and don't forgot to clean up on error and on success.